inst/tests/tests.Rraw

```diff
@@ -3732,12 +3732,15 @@ test(1100, dt1[dt2,roll=-Inf,rollends=c(FALSE,TRUE)]$ind, INT(NA,NA,1,2,2,2,2,2,
 
   # test for #5512, only compute default fill if needed.
   DT = data.table(chr=c("a","b","b"), int=1:3, dbl=as.double(4:6))
-  test(1102.39, dcast(DT, . ~ chr, min, value.var="int"), data.table(.=".",a=1L,b=2L,key="."))#fill not used in output, so default fill not computed, and no warning emitted.
+  mymin <- function(x){
+    if(length(x)==0)stop("calling mymin on vector of length 0")
```

Comment:
There are two operations which are relevant to test, (1) computing the default fill value by using fun.aggregate on a zero-length vector, and (2) coercing the result to the output type.
When fun.aggregate=min with integer(), we get a warning `NAs introduced by coercion to integer range` which comes from (2) whereas the other warning `no non-missing arguments to min; returning Inf` which would happen in (1) is suppressed in R code:
```r
fill.default = suppressWarnings(dat[0L][, eval(fun.call)])
```
But the point of this PR is that it would be preferable to do neither (1) nor (2) in the case where there are no missing entries to fill, so this is why I add tests involving mymin, which stops with an error if it is called on a zero-length vector.

Reply:
PR making much more sense now, I had misunderstood -- I thought we were _also_ avoiding warning for a case like

```r
dcast(DT, a ~ b, value.var = "c", fill = 'xxx')
```

Where `c` is numeric. I.e., if `fill` is not needed we don't need the `as.numeric('xxx')` warning to show up.

Indeed, IINM that _does_ happen on this PR:

```r
DT = data.table(a = 1, b = 2, c = 3)
# warns on master, not on this branch
dcast(DT, a ~ b, value.var = 'c', fill = 'xxx')
```

Was that an unintentional improvement? If so we could add a test to make sure it stays

Reply:
yes indeed I added a test.